internal/uidriver/glfw/ui_unix.go

```diff
@@ -19,10 +19,89 @@
 package glfw
 
 import (
+	"math"
+
 	"github.com/hajimehoshi/ebiten/v2/internal/driver"
 	"github.com/hajimehoshi/ebiten/v2/internal/glfw"
+	"github.com/jezek/xgb"
+	"github.com/jezek/xgb/randr"
+	"github.com/jezek/xgb/xproto"
 )
 
+type videoModeScaleCacheKey struct{ X, Y int }
+
+var videoModeScaleCache map[videoModeScaleCacheKey]float64
```

Comment:
Let's clear the cache at updateMonitors (a handler of Monitor event), or wrong values are cached when monitors are dynamically changed. (Actually there is a TODO for `devicescale` packages.)

Reply:
Doing that for now; in the future, maybe we should remove use of GetVideoMode() calls entirely and route all that through *monitor.

Leaving as is for now, though, as that's a separate refactor and also a bit unclear if a good idea (as that'd likely mean to query XRandR for _every_ monitor on monitor change, while usually we will only query for the one the window is on).

Reply:
Cache invalidation added.